### PR TITLE
Bump `dfx` to `0.14.0` and enable wasm optimizer

### PR DESCRIPTION
**Pull request by kentosugama**

### CODE REVIEW

**Comment by kentosugama** [comment]
`wasm-opt` is scrambling internal function names, which is messing with the `wasm_profiling()` primitive in `ic-repl` that specifies which function to profile.

**Comment by chenyan-dfinity** [comment]
Ah right. The name section is missing after wasm-opt. Maybe we can disable wasm-opt for the GC benchmarks?

**Comment by kentosugama** [comment]
Just ran a test. The names are still in the flamegraphs

File: collections/README.md [hunk 1]
```diff
@@ -28,3 +28,4 @@ the same elements, and the queries are exactly the same. Below we explain the me
 > * `btree` comes from [Byron Becker's stable BTreeMap library](https://github.com/canscale/StableHeapBTreeMap).
 > * `zhenya_hashmap` comes from [Zhenya Usenko's stable HashMap library](https://github.com/ZhenyaUsenko/motoko-hash-map).
 > * The MoVM table measures the performance of an experimental implementation of Motoko interpreter. External developers can ignore this table for now.
+> * The optimizer in `dfx` is disabled for the garbage collection cases because the optimizer removes the name section, making profiling trickier.
```
Comment (chenyan-dfinity) [inline collections/README.md:31]: This should go under motoko/README.md, right?

Comment (kentosugama) [reply collections/README.md]: Yes, thank you

**Comment by kentosugama** [comment]
Hey Yan, I'm just getting back to this now.

The benchmark change in the garbage collector cases seem to be from the bump in `dfx` version.
See here: https://github.com/dfinity/canister-profiling/pull/53
Not quite sure why yet.

Will look into the name section next.

**Comment by kentosugama** [comment]
Update: #54 should resolve the GC cases anomalies

**Comment by kentosugama** [comment]
https://github.com/dfinity/ic-wasm/pull/34
Added a change to preserve the name sections for profiling.

This changed is used in #55. Once #55 is merged, I will close this PR.

**Pull request closed without merging**